src/ynEIGEN/EigenStrategyManager.sol

```diff
@@ -0,0 +1,352 @@
+    }
+
+    function toEigenLayerDeposit(
+        IERC20 asset,
+        uint256 amount
+    ) internal returns (IERC20 depositAsset, uint256 depositAmount) {
+        if (address(asset) == address(wstETH)) {
+            // Adjust for wstETH
+            depositAsset = IERC20(wstETH.stETH());
+            depositAmount = wstETH.unwrap(amount); 
+        } else if (address(asset) == address(woETH)) {
+            // Adjust for woeth
+            depositAsset = IERC20(woETH.asset()); 
+            // calling redeem with receiver and owner as address(this)
+            depositAmount = woETH.redeem(amount, address(this), address(this)); 
+        } else {
+            // No adjustment needed
+            depositAsset = asset;
+            depositAmount = amount;
+        }   
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ADMIN  -------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Adds a new strategy for a specific asset.
+     * @param asset The asset for which the strategy is to be added.
+     * @param strategy The strategy contract address to be associated with the asset.
+     */
+    function addStrategy(IERC20 asset, IStrategy strategy)
+        external
+        onlyRole(STRATEGY_ADMIN_ROLE)
+        notZeroAddress(address(asset))
+        notZeroAddress(address(strategy)) {
+        if (address(strategies[asset]) != address(0)){
+            revert StrategyAlreadySetForAsset(address(asset));
+        }
+
+        strategies[asset] = strategy;
+        emit StrategyAdded(address(asset), address(strategy));
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  VIEWS  -------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Retrieves the total balances of staked assets across all nodes.
+     * @param assets An array of ERC20 tokens for which balances are to be retrieved.
+     * @return stakedBalances An array of total balances for each asset, indexed in the same order as the `assets` array.
+     */
+    function getStakedAssetsBalances(IERC20[] calldata assets) public view returns (uint256[] memory stakedBalances) {
+
+        stakedBalances = new uint256[](assets.length);
+        // Add balances contained in each TokenStakingNode, including those managed by strategies.
+
+        ITokenStakingNode[] memory nodes = tokenStakingNodesManager.getAllNodes();
+        uint256 nodesCount = nodes.length;
+        uint256 assetsCount = assets.length;
+        for (uint256 j = 0; j < assetsCount; j++) {      
+
+            IERC20 asset = assets[j];
+            for (uint256 i; i < nodesCount; i++ ) {
```

Comment:
Seeing no limits to the number of nodes, I would like to know how much gas you expect this call to cost and how much node/assets you plan to use.

Reply:
Good question we plan to do create 5-10 nodes max, and then make a new release where we optimize this computation